### ISSUE TITLE
False positive in `commented_code_linter()`?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1469]
I wouldn't have expected this to produce a lint:

``` r
library(lintr)

lint(
  text = "# Here we make sure that both atomic and non-atomic vector arguments work.
          # That is, users can do any of the following:
          # - `x = c(...)`
          # - `x = list(...)`",
  linters = commented_code_linter()
)
#> <text>:3:13: style: [commented_code_linter] Commented code should be removed.
#>           # - `x = c(...)`
#>             ^~~~~~~~~~~~~~
#> <text>:4:13: style: [commented_code_linter] Commented code should be removed.
#>           # - `x = list(...)`
#>             ^~~~~~~~~~~~~~~~~
```

<sup>Created on 2022-07-25 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

**Comment by MichaelChirico** [comment]
the issue is that anything inside \` is just a variable name, so this parses as `-x` where `x` is a variable named `"x = c(...)"`...

let's track this recurring issue with `commented_code_linter()` in #934

**Comment by IndrajeetPatil** [comment]
Sounds good, thanks!
